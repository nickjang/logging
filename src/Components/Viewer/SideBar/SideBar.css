.sidebar {
  display: flex;
  flex-direction: row;
  height: calc(100vh - 80px);
  position: absolute;
  background-color: white;
  z-index: 1;
  box-shadow: 10px 0px 15px -4px rgba(0, 0, 0, 0.1), 4px 0px 7px -2px rgba(0, 0, 0, 0.05);
}

.sidebar-container {
  width: 100%;
  overflow: auto;
}

.sidebar-main {
  width: fit-content;
  height: 100%;
  padding-bottom: 300px;
}

.sidebar button.open-sidebar {
  font-family: 'Do Hyeon', sans-serif;
  font-size: 14px;
  min-width: 30px;
  border: 0;
  color: darkgray;
  background-color: white;
}

.sidebar .form-status {
  display: inline-block;
}

.sidebar .note {
  max-width: fit-content;
  width: 100%;
  margin: 1.75rem 1rem auto 1rem;
}

.sidebar .form-status {
  margin-left: 1rem;
  margin-top: 1rem;
}

.sidebar .sidebar-top-btn {
  margin: 2rem 0rem 0 1rem;
}

.sidebar .side-bar-projects {
  list-style: none;
  padding: 0;
}
