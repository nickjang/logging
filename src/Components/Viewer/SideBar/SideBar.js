import React, { Component } from 'react';
import ProjectPicker from '../ProjectPicker/ProjectPicker';

class SideBar extends Component {
  state = {
    loading: false,
    fetchError: {
      code: '',
      message: ''
    }
  }

  // fetchProjectsORAllProjectsAndLogs() {}
  // if user enters site from viewpage, 
  // fetch project or all projects and logs to load sidebar

  // fetchLogs() {}
  // if all logs not loaded in App already, see handlesubmit below

  /* if not loading all logs, fetch logs on submit*/
  handleSubmit = (e) => {
    e.preventDefault();
    //loading/error
    //use picked context to store picked, and fetch logs of picked;

    //otherwise, remove handlesubmit and view selected button,
    //and put picked context in viewmain page and store picked there
    //and update main page on picked update.
    //--------------------------------------------------
    //if store all projects and logs in app, add newly created logs to inital fetch of all logs
    //and pass all projects and logs to view main page,
    //but also passes and stores all projects and logs to log main page

    //if store just projects in app, fetch newly created logs when clicking logmain page,
    //and include viewselectedbutton and fetch logs when clicking view button in sidebar,
    //but need to fetch logs each time click view and can't deactivate days there's no logs
  }

  render() {
    const projects = this.props.projects.map(project => 
      <ProjectPicker key={project.id} />
    );
    return (
      <aside class='sidebar'>
        <output form='sidebar-form' className='form-status'>{this.state.fetchError || (this.state.loading && 'Loading projects...')}</output>
        {/* fetch projects if not already fetch on log in -- but will log in be saved on refresh?*/}
        {/* also for getting logs if all logs not stored in state*/}
        <p class='note'>View logs from selected projects or their year(s), month(s), and day(s).</p>
        <button type='submit' form='sidebar-form' onClick={(e) => { this.handleSubmit(e) }}>View Selected</button>
        <p class='note'>Overlapping logs will be counted once</p>
        <form action='' id='sidebar-form'>
          {/*<!--add scroll so view button stays in view-->*/}
          <ul>
            {projects}
          </ul>
        </form >
      </aside >
    );
  }
}

export default SideBar;