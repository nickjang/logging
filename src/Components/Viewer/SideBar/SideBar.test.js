import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SideBar from './SideBar';

describe(`SideBar component`, () => {
  it('renders SideBar by default', () => {
    const wrapper = shallow(<SideBar />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})