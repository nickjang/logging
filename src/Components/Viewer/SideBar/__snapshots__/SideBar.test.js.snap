// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component renders SideBar by default 1`] = `
<aside
  className="sidebar opened"
>
  <div
    className="sidebar-container"
  >
    <div
      className="sidebar-main"
    >
      <p
        className="note"
      >
        View logs from selected projects or their year(s), month(s), and day(s).
      </p>
      <p
        className="note"
      >
        Overlapping selections will be counted once. Calendars are displayed in your time zone. Logs will also displayed in your time zone.
      </p>
      <button
        className="sidebar-top-btn lg-btn lg-btn-success"
        disabled={true}
        form="sidebar-form"
        onClick={[Function]}
        type="submit"
      >
         View logs
      </button>
      <button
        className="sidebar-top-btn lg-btn"
        form="sidebar-form"
        onClick={[Function]}
        type="reset"
      >
         Clear selections
      </button>
      <output
        className="form-status "
        form="sidebar-form"
      >
         
      </output>
      <form
        action=""
        id="sidebar-form"
      >
        <ul
          className="side-bar-projects mt-1"
        >
          <li
            className="note"
          >
            No projects to make selections from
          </li>
        </ul>
      </form>
    </div>
  </div>
  <button
    className="open-sidebar"
    onClick={[Function]}
  >
     
    &lt;
  </button>
</aside>
`;
